### ISSUE TITLE
Discriminator is not training properly?

### ISSUE THREAD

**Issue by ThisisBillhe** [FoundationVision/LlamaGen#27]
Hi peize, I try to train VQGAN with your default config, where the discriminator start training after 20k iterations. However, I notice that logits_real and logits_fake are very close all the time. For example:
`Beginning epoch 7...
 (Generator) rec_loss: 0.0441, perceptual_loss: 0.2432, vq_loss: 0.0107, commit_loss: 0.0027, entropy_loss: -0.0000, codebook_usage: 0.9761, generator_adv_loss: 0.0698, disc_adaptive_weight: 1.0000, disc_weight: 0.5000
(Discriminator) discriminator_adv_loss: 0.4980, disc_weight: 0.5000, logits_real: -0.1318, logits_fake: -0.1396`

To be a good discriminator, logits_real should be close to 1, while logits_fake should be close to -1, right? Can you share me your training log regarding these logits?

**Comment by PeizeSun** [comment]
Hi~
This is exactly what the discriminator should be. 

If logits_real is very close to 1, and logits_fake is vey close to -1, the adversarial process is dominated by discriminator, generator can't benefit from discriminator.

**Comment by ThisisBillhe** [comment]
Hi peize, 

Thanks for your reply! I understand that the discriminator will be too strong if logits_real is close to 1, and logits_fake is close to -1. But shouldn't they be one positive and the other negative? If they have very similar values (like logits_real: -0.1318, logits_fake: -0.1396), to my understanding, the discriminator can not discriminate at all?

**Comment by PeizeSun** [comment]
The beauty of adversarial process is that logits_fake and logits_real are similar.
I highly recommend to run other GAN code to see this magic phenomena.

**Comment by ThisisBillhe** [comment]
Thanks for your help!

**Comment by Doctor-James** [comment]
I would like to ask if the discriminator's logit_real and logit_fake are always similar, and when they are both positive, the discriminator_adv_loss will be negative. Is such a discriminator beneficial to the generator?
